app.js

```diff
@@ -70,18 +70,58 @@ bot.on('text', (ctx) => {
             if(textMsg.toLowerCase().search('mybatchmates') >= 0) {
                 //get my batchmates
                 dbo.collection(config.get('mongoCollections.users')).findOne({"from.id" : fromId}, function(error, user) {
-                    var reply = "Here are your batchmates : \n";
-                    dbo.collection(config.get('mongoCollections.users')).find({"batch" : user.batch}).forEach(function(batchmate) {
+                    var reply = `Here are your batchmates : \n`;
+                    dbo.collection(config.get('mongoCollections.users')).find({"batch" : user.batch}).toArray( function(err , batchmates) {
+                        batchmates.map(batchmate => {
+                            console.log(batchmate)
+                            reply += batchmate.from.first_name ;
+                            if( batchmate.from.last_name) {
+                                reply +=  `${batchmate.from.last_name} \n`;
+                            } 
+                        })
+                        utility.sendMessage(ctx, reply, 'search', 'private'); 
+                    })
+                })
+            } else if(textMsg.toLowerCase().search('batch_') >= 0) {
+                //get batchmates of given year { pattern is /batch_YYYY }
+                var year = textMsg.substring(textMsg.length - 4, textMsg.length); 
+                console.log(year)
+                var reply = `Here are batchmates from ${year} : \n`;
+                dbo.collection(config.get('mongoCollections.users')).find({"batch" : year}).toArray( function(err , batchmates) {
```

Comment:
could use regex here too, as I have observed, some people haven't stuck to the format of input :))

Reply:
@saumye1 Done !!!!